File: test/duckdb/check-db.js
// 检查数据库中的数据
import duckdb from 'duckdb';

const dbPath = 'D:/dev/ai/ivydata/db/ivy.duckdb';

async function checkDatabase() {
  try {
    console.log('连接到数据库:', dbPath);
    const db = new duckdb.Database(dbPath);
    
    // 检查表是否存在
    db.all('SELECT name FROM sqlite_master WHERE type="table"', (err, tables) => {
      if (err) {
        console.error('检查表失败:', err);
        db.close();
        return;
      }
      
      console.log('数据库中的表:', tables);
      
      // 检查base_data表中的数据
      db.all('SELECT COUNT(*) as count FROM base_data', (err, result) => {
        if (err) {
          console.error('检查数据失败:', err);
          db.close();
          return;
        }
        
        console.log('base_data表中的数据行数:', result[0].count);
        
        // 查看前10条数据
        db.all('SELECT * FROM base_data LIMIT 10', (err, rows) => {
          if (err) {
            console.error('查看数据失败:', err);
            db.close();
            return;
          }
          
          console.log('前10条数据:', rows);
          db.close();
        });
      });
    });
  } catch (error) {
    console.error('连接数据库失败:', error);
  }
}

checkDatabase();

File: test/duckdb/check-db.py
import duckdb

# 连接到数据库
DB_PATH = r'D:\dev\ai\ivydata\db\ivy.duckdb'

print(f'连接到数据库: {DB_PATH}')
conn = duckdb.connect(DB_PATH)

# 检查表是否存在
print('\n检查数据库中的表:')
tables = conn.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'main'").fetchall()
print(f'表列表: {tables}')

# 检查表结构
print('\n检查base_data表结构:')
columns = conn.execute('PRAGMA table_info(base_data)').fetchall()
for column in columns:
    print(f'列名: {column[1]}, 类型: {column[2]}, 非空: {column[3]}, 默认值: {column[4]}, 主键: {column[5]}')

# 检查数据行数
print('\n检查base_data表中的数据行数:')
count = conn.execute('SELECT COUNT(*) as count FROM base_data').fetchone()
print(f'数据行数: {count[0]}')

# 查看前10条数据
print('\n查看前10条数据:')
rows = conn.execute('SELECT * FROM base_data LIMIT 10').fetchall()
for row in rows:
    print(row)

# 关闭连接
conn.close()
print('\n连接已关闭')
